src/main/scala/nl.knaw.dans.easy.deposit/EasyDepositApiApp.scala

```diff
@@ -262,9 +264,13 @@ class EasyDepositApiApp(configuration: Configuration) extends DebugEnhancedLoggi
    * @param is   the input stream to write from
    * @return `true` if a new file was created, `false` otherwise
    */
-  def writeDepositFile(is: => InputStream, user: String, id: UUID, path: Path): Try[Boolean] = {
+  def writeDepositFile(is: => InputStream, user: String, id: UUID, path: Path, contentType: String): Try[Boolean] = {
     for {
+      _ <- if (Option(contentType).toSeq.exists(!_.trim.matches(contentTypeZipPattern))) Success(())
+           else Failure(BadRequestException("Content-Type must not be application/zip."))
       dataFiles <- getDataFiles(user, id)
+      _ <- if (!(dataFiles.bag / "data" / path.toString).isDirectory) Success(())
```

Comment:
if you swap the if-else branches, your code becomes a little more easy to read.

Or perhaps even a better idea: extract method for both these if-else constructs that you added?